start

```diff
@@ -27,7 +27,8 @@ echo "Starting karen..."
 echo
 ./karen &
 KAREN_PID=$!
-echo $! > ./karen.pid
+KAREN_PGID=$(ps opgid= "$KAREN_PID")
```

Comment:
Do we want to do this?

Not tested but doesn't this mean if we do a shutdown then we run `./stop` to kill karen and all it's child processes including the shutdown child process and then prevent the shutdown?

Not sure if shutdown works like that. Maybe at that point the shutdown process is already being handled by the kernel so killing `shutdown` doesn't do anything, I'm not sure. Needs testing though.

Reply:
Ah, you're right. The shutdown doesn't work if karen and her children are killed lol.

The goal was to not spawn multiple instances of karen. And I think you're right, the approach of killing karen isn't right and will break not only shutdown and reboot, but also the OTA updates. 

We should probably add a check in karen itself to not run if it's already running.

Reply:
>We should probably add a check inside karen itself to not run if it's already running.

Yeah definitely, that sounds like a good approach.

We can write a pidfile inside `karen` to the events dir it's watching, then remove it when the script exits with `trap`.

Reply:
Yes, but then we'll also have to check and remove that file on boot as it can exist after a power failure/crash. Perhaps we can just check against `ps -A | grep karen` (inside karen, before starting up) for now...

Reply:
That's true, `ps -A | grep karen` won't work because it's actually `bash ./karen` not `karen`. You can do `ps -Af | grep karen` but there is a higher potential for false positives.

e.g `someprocess --that-is-definitely-no-karen` will return as a match for `ps -Af | grep karen`.

I think it should be fine though.

Reply:
Ah okay, understood. Sounds good, opened a separate issue for it #57